### PR TITLE
Make find test under cursor handle namespaced def macros

### LINKED ISSUE #1086: Calva not detecting tests with aliased `clojure.test` namespace
Calva is unable to detect tests if created with aliased `t` namespace instead of direct `refer`:

```clojure
(ns ...
      (:require ...
            [clojure.test :refer [deftest is] :as t]))

(deftest foo
  (is (= 1 1)))

; Running test: foo…
; 1 tests finished, all passing 👍, ns: 1, vars: 1

(t/deftest bar
  (t/is (= 1 1)))

; No test found at cursor
```

### PR DESCRIPTION
**Pull request by PEZ**
## What has Changed?

We had some pretty complicated way of finding ”the current test”, which also fails when using namespace qualified definition macros. Fixing that here, reusing the ”current top level definied thing” that we use for custom code snippets. It besically disregards whatever is in the function position of the toplevel form and just picks the next non-meta symbol. I can't see when this should be worse than what we previously did, and it certainly is better for the namspace qualified macro case.

Fixes #1086

## My Calva PR Checklist

I have:

- [x] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [x] Directed this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [x] Made sure I have changed the PR base branch, so that it is not `published`. (Sorry for the nagging.)
- [x] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.
- [x] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and either tested it there if so, or mentioned it in the PR.
- [ ] Tested the VSIX built from the PR (so, after you've submitted the PR). You'll find the artifacts by clicking _Show all checks_ in the CI section of the PR page, and then _Details_ on the `ci/circleci: build` test. NB: *There is a CircleCI bug that makes the Artifacts hard to find. Please see [this issue](https://discuss.circleci.com/t/artifacts-tab-not-showing-unless-logged-in/32433) for workarounds.*
     - [x] Tested the particular change
     - [x] Figured if the change might have some side effects and tested those as well.
     - [x] Smoke tested the extension as such.
- [x] Referenced the issue I am fixing/addressing _in a commit message for the pull request_.
     - [x] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)

Ping @bpringe

### CODE REVIEW

**Comment by bpringe** [comment]
I noticed that `Run Current Test` on the `with-test` macro does not run the test with this vsix, but it does in current Calva.

```clojure
(with-test
  (defn foo []
    #{{:id           1010101
       :display-name "Campaign 1010101"
       :state        "enabled"}})
  (is (= #{{:id 1010101
            :display-name "Campaign 1010101"
            :state "enabled"}}
         (foo))))
```

Output:
```clojure
; Running test: defn…
; No tests found. 😱, ns: 0, vars: 0
```

**Comment by PEZ** [comment]
Good catch!

**Comment by PEZ** [comment]
Can you test this again, @bpringe ? And extra points if you have a look at the code again, of course. 😄

File: src/util/get-text.ts [hunk 1]
```diff
@@ -41,18 +42,11 @@ export function currentFunction(editor: vscode.TextEditor): SelectionAndText {
...
-                const symbol = tokenCursor.getToken();
-                if (symbol.type === 'id') {
-                    return [selection, symbol.raw];
-                }
-            }
+        const cursorOffset = document.offsetAt(editor.selection.active);
+        const tokenCursor = docMirror.getDocument(document).getTokenCursor(cursorOffset);
+        const [range, text] = cursorTextGetter.currentTopLevelFunction(tokenCursor);
+        if (range) {
+            return [select.selectionFromOffsetRange(document, range), text]; // TODO: Figure out which selection to return...
```
Comment (bpringe) [inline src/util/get-text.ts]: I don't understand this TODO. Is it still a todo? Looks like there's only one selection to return.

**Review by bpringe — APPROVED** [review]
Looks good to me. Just had a question about a code comment.

**Comment by bpringe** [comment]
Also, I tested it and it works fine for me.

File: src/util/get-text.ts (hunk 1 above)
Comment (PEZ) [reply src/util/get-text.ts]: Thanks. Yeah, we were previously returning the text of the function name, but the selection of the whole function. Which seemed really weird, but I didn't know if that was intentional. I don't think it was, and for this particular function no-one is using the selection so I fixed it. And forgot the TODO. 😄

**Pull request merged**